File: backend/app/api/comments.py
from flask import jsonify, request
from bson import ObjectId
from app.models import Task, Comment
from app.api import bp

@bp.route('/tasks/<task_id>/comments', methods=['GET'])
def get_comments(task_id):
    try:
        object_id = ObjectId(task_id)
    except:
        return jsonify({'error': 'Invalid task ID'}), 400
        
    task = Task.get_or_404(object_id)
    return jsonify([comment.to_dict() for comment in task.comments])

@bp.route('/tasks/<task_id>/comments', methods=['POST'])
def create_comment(task_id):
    try:
        object_id = ObjectId(task_id)
    except:
        return jsonify({'error': 'Invalid task ID'}), 400
        
    task = Task.get_or_404(object_id)
    data = request.get_json() or {}
    
    if 'content' not in data:
        return jsonify({'error': 'Missing required field: content'}), 400
    
    comment = Comment(
        task_id=task_id,
        content=data['content']
    )
    comment.save()
    return jsonify(comment.to_dict()), 201

@bp.route('/comments/<comment_id>', methods=['GET'])
def get_comment(comment_id):
    try:
        object_id = ObjectId(comment_id)
    except:
        return jsonify({'error': 'Invalid comment ID'}), 400
        
    comment = Comment.get_or_404(object_id)
    return jsonify(comment.to_dict())

@bp.route('/comments/<comment_id>', methods=['PUT'])
def update_comment(comment_id):
    try:
        object_id = ObjectId(comment_id)
    except:
        return jsonify({'error': 'Invalid comment ID'}), 400
        
    comment = Comment.get_or_404(object_id)
    data = request.get_json() or {}
    
    if 'content' not in data:
        return jsonify({'error': 'Missing required field: content'}), 400
    
    comment.content = data['content']
    comment.save()
    return jsonify(comment.to_dict())

@bp.route('/comments/<comment_id>', methods=['DELETE'])
def delete_comment(comment_id):
    try:
        object_id = ObjectId(comment_id)
    except:
        return jsonify({'error': 'Invalid comment ID'}), 400
        
    comment = Comment.get_or_404(object_id)
    comment.delete()
    return '', 204

File: src/apps/frontend/api/comments.js
import { apiClient } from './client';

/**
 * Get comments for a specific task
 * @param {string} taskId - The ID of the task
 * @returns {Promise} Response containing array of comments
 */
export const getTaskComments = async (taskId) => {
    return await apiClient.get(`/tasks/${taskId}/comments`);
};

/**
 * Create a new comment for a task
 * @param {string} taskId - The ID of the task
 * @param {Object} data - Comment data
 * @param {string} data.content - The content of the comment
 * @returns {Promise} Response containing the created comment
 */
export const createComment = async (taskId, data) => {
    return await apiClient.post(`/tasks/${taskId}/comments`, data);
};

/**
 * Update an existing comment
 * @param {string} commentId - The ID of the comment to update
 * @param {Object} data - Updated comment data
 * @param {string} data.content - The new content of the comment
 * @returns {Promise} Response containing the updated comment
 */
export const updateComment = async (commentId, data) => {
    return await apiClient.put(`/comments/${commentId}`, data);
};

/**
 * Delete a comment
 * @param {string} commentId - The ID of the comment to delete
 * @returns {Promise} Response indicating success/failure
 */
export const deleteComment = async (commentId) => {
    return await apiClient.delete(`/comments/${commentId}`);
};